services/api/src/resources/notification/resolvers.ts

```
@@ -711,9 +711,43 @@ export const getAllNotifications: ResolverFn = async (
   args,
   { sqlClientPool, hasPermission }
 ) => {
-  await hasPermission('notification', 'viewAll');
+  let rows;
+
+  if (args.name && args.type) {
+    rows = await query(sqlClientPool, Sql.selectNotificationByNameAndType(args.name, args.type));
+
```

need to throw an error if no rows

---

or return `null` or something